### DISCUSSION TITLE
How to display Error message with Toast and Redirect using Server Action

### DISCUSSION THREAD

**Discussion by surnr** [vercel/next.js#57118] [Help]
### Summary

```js
"use server"

import { redirect } from "next/navigation";

export async function SignUpNewUser(email) {
      const userExist = User.findOne({email});
      if (userExist) {
              return {success: failed, error: "User already exists" }
              redirect("/login")
      }
}
```

How to show the error of the user already exists and redirect him to the login page,? when I use the return JSON response, the redirect won't work.

Thanks for the help mate

**Comment by KryptXBSA** [comment]
One option is that you can redirect to `/login?error=userExists` and use `searchParams` page prop to check if there is an error.

**Comment by chungweileong94** [comment]
You could set cookie before redirect, then read the cookie and show the toast accordingly.
